services/gateway/gateway-service.mjs

```diff
@@ -185,15 +185,34 @@ class GatewayService {
     }
 
     // Synchronize the state.
-    let firstLocation = {playerID: playerID, 
+    const firstLocation = {playerID: playerID,
       displayName: socket.playerData.displayName};
     firstLocation.mapCoord = socket.playerData.mapCoord;
+    if (firstLocation.mapCoord === undefined) {
+      let setLocation = false, index = 0;
+      let spawnPoint = this.gameMap.getSpawnPoint();
+      for (let i = spawnPoint.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        const temp = spawnPoint[i];
+        spawnPoint[i] = spawnPoint[j];
+        spawnPoint[j] = temp;
+      }
+      while (setLocation === false && index < spawnPoint.length) {
+        firstLocation.mapCoord = spawnPoint[index];
+        setLocation = await this._occupyCoord(firstLocation.mapCoord, playerID);
+        index = index + 1;
+      }
+      if (setLocation === false) {
+        console.warn(`No free spawn point.`);
+        socket.disconnect();
+        return;
+      }
```

Comment:
You had better add comments if other people cannot get what the code does at first glance.

Reply:
Add some comments.